src/smc-webapp/editor.coffee

```diff
@@ -1085,8 +1085,11 @@ class CodeMirrorEditor extends FileEditor
                 that.click_edit_button($(@).data('name'))
                 return false
 
+        # FIXME: removing the button here is pointless, since it will be added later again (somewhere)
         # FUTURE: implement printing for other file types
-        if @filename.slice(@filename.length-7) != '.sagews'
+        is_sagews = @filename.slice(@filename.length-7) != '.sagews'
```

Comment:
What's going on here?  

1. The name `is_sagews` suggess it should be `@filename.slice(@filename.length-7) == '.sagews'` instead of `@filename.slice(@filename.length-7) != '.sagews'`?

2. Can you use use filename_extension from misc, rather than the hardcoded length of 7, which is a little more dangerous code?